yarpctest/recorder/recorder_test.go

```diff
@@ -126,3 +133,102 @@ func TestHash(t *testing.T) {
 	b := []byte(rgen.Atom())
 	assert.NotEqual(t, recorder.hashRequest(&r, b), referenceHash)
 }
+
+func TestOverwriteReplay(t *testing.T) {
+	dir, err := ioutil.TempDir("", "yarpcgorecorder")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	func() {
+		// First we double check that our cache is empty.
+		recorder := NewRecorder(t, Option{
+			Mode:        Replay,
+			RecordsPath: dir,
+		})
+
+		clientDisp := yarpc.NewDispatcher(yarpc.Config{
+			Name: "client",
+			Outbounds: transport.Outbounds{
+				"server": http.NewOutbound("http://localhost:65535"),
+			},
+			Filter: recorder,
+		})
+		require.NoError(t, clientDisp.Start())
+		defer clientDisp.Stop()
+
+		client := raw.New(clientDisp.Channel("server"))
+		ctx, _ := context.WithTimeout(context.Background(), time.Second)
+
+		_, _, err := client.Call(ctx, yarpc.NewReqMeta().Procedure("hello"), []byte("Hello"))
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "no such file or directory")
```

Comment:
"no such file or directory" is not helpful in telling the user that a record doesn't exist for the request. As mentioned in the [other message](https://github.com/yarpc/yarpc-go/pull/365#discussion_r82091443) about `IsNotExist`, check that error right after `ioutil.ReadFile` and provide a better error message. Something like,

> Request to [...] is not recorded and the current recording mode [..] prevents recording new requests. Pass `--recorder append` or `--recorder overwrite` to record new requests.